### DISCUSSION TITLE
Debugging Plug-ins in Visual Studio Code

### DISCUSSION THREAD

**Discussion by LHBL2003** [netbox-community/netbox#9397] [Q&A]
I've been friends with NetBox for a couple of weeks. unfortunately the plugins that interest me are not really maintained. I would like to improve or extend some plugins. I was able to correct one or the other error via text file. Unfortunately, it's not easy to extend someone else's plugin or fix a bug once you have to find the entry point. Is there a guide how to B. with Visual Studio Code or co. Debug the plugin code at runtime?

It is particularly important to me to debug the Python code.

----------------------------

I don't know if I'm already on the right track.
Since it doesn't show the web page I expect at 8000, I think the text during installation showed the Netbox page.  I would expect the same now when debugging.

**I have tried the following so far:**

- systemctl stop nginx
- systemctl stop netbox
- Opened Visual Studio Code
- Installed Pyton for VSCode
- pip install Django==4.0.4
- Clone a GitHub plugin project
- In Terminal Window add: source /opt/netbox/venv/bin/activate
- Python file opened (models.py)
- In VSCode clicked on Python in the lower right corner and selected Interpreter Path with the following path: (/opt/netbox/venv/bin/python3)
- Left click on Debug and select "Chreate a launch .json file".
- Debug Configuration: "Django Launch and debug a Diango web application" selected
- For manage.py selected the path "opt/netbox/netbox/manage.py
- Start Debugging Python: Django

<img width="1790" alt="image" src="https://user-images.githubusercontent.com/46369917/169397164-fae42716-7600-4180-bd33-fcce1c2d907f.png">

<img width="1413" alt="image" src="https://user-images.githubusercontent.com/46369917/169397535-805962c7-49ae-47b5-b07e-8e11f2e1bdd7.png">

-------------------------------------------

If I stop debugging and try the following instead:

Terminal:
source /opt/netbox/venv/bin/activate
python3 manage.py runserver 0.0.0:8000 --insecure

Then I get NetBox via https://127.0.0.1:8000

I think I am missing only a small configuration for VSCode?

**Comment by kkthxbye-code** [comment]
Try adding --insecure to the args in your launch.json

```
"args": [
  "runserver",
  "--insecure"
]
```

Something like that.
